ironic/resource_ironic_node_v1.go

```diff
@@ -585,3 +622,126 @@ func changePowerState(client *gophercloud.ServiceClient, d *schema.ResourceData,
 
 	return nil
 }
+
+// setRAIDConfig calls ironic's API to send request to change a Node's RAID config.
+func setRAIDConfig(client *gophercloud.ServiceClient, d *schema.ResourceData) (err error) {
+	var logicalDisks []nodes.LogicalDisk
+	var raid *metal3v1alpha1.RAIDConfig
+
+	raidConfig := d.Get("raid_config").(string)
+	if raidConfig == "" {
+		return nil
+	}
+
+	err = json.Unmarshal([]byte(raidConfig), &raid)
+	if err != nil {
+		return
+	}
+
+	err = checkRAIDConfigure(d.Get("raid_interface").(string), raid)
+	if err != nil {
+		return
+	}
+
+	// Build target for RAID configuration steps
+	logicalDisks, err = ironic.BuildTargetRAIDCfg(raid)
+	if len(logicalDisks) == 0 || err != nil {
+		return
+	}
+
+	// Set root volume
+	if len(d.Get("root_device").(map[string]interface{})) == 0 {
+		logicalDisks[0].IsRootVolume = new(bool)
+		*logicalDisks[0].IsRootVolume = true
+	} else {
+		log.Printf("rootDeviceHints is used, the first volume of raid will not be set to root")
+	}
+
+	// Set target for RAID configuration steps
+	return nodes.SetRAIDConfig(
+		client,
+		d.Id(),
+		nodes.RAIDConfigOpts{LogicalDisks: logicalDisks},
+	).ExtractErr()
+}
+
+// buildBIOSSettings builds bios settings for BIOS clean steps
+func buildBIOSSettings(d *schema.ResourceData, firmwareConfig *bmc.FirmwareConfig) (settings []map[string]string, err error) {
+	acc, err := bmc.NewAccessDetails(d.Get("bmc_address").(string), d.Get("bmc_disable_certificate_verification").(bool))
```

Comment:
As mentioned above, I think the AccessDetails should be derived from driver_info instead of adding these new fields?

Reply:
I understand your concerns, but the keys of `DisableCertificateVerification` corresponding to different drivers in `driver_info` are different (eg. irmc: irmc_verify_ca; idrac: drac_verify_ca). Although they all use `_verify_ca` as the suffix, it seems we can base on this to get the value of `DisableCertificateVerification`, but IMO it is unreliable.

Reply:
I think it is unreliable because I didn't see a clear agreement that the key of `DisableCertificateVerification` in driver_info must be named in the form of `*_verify_ca`. But it seems that everyone abides by this rule by default, so may the `DisableCertificateVerification` can be derived from driver_info. But `BMC.Address` cannot, as the prefix of the address is not recorded in the driver_info. If you have a better way to derive `DisableCertificateVerification` from driver_info, please let me know.

Reply:
Ugh I see what you mean :(

For the benefit of other reviewers, the installer creates an [AccessDetails object](https://github.com/openshift/installer/blob/master/pkg/tfvars/baremetal/baremetal.go#L78), but then transforms that into a [DriverInfo structure](https://github.com/openshift/installer/blob/master/pkg/tfvars/baremetal/baremetal.go#L86) which is passed into terraform.

Because the DriverInfo contains [driver-specific keys](https://github.com/metal3-io/baremetal-operator/blob/master/pkg/hardwareutils/bmc/irmc.go#L63) we end up with e.g `irmc_verify_ca` in the `driver_info` which can't easily be recovered into the original flag.

It seems like we ideally need e.g `bmc.AccessDetailsFromDriverInfo` - or some other helper that abstracts the driver specific keys :(

@dtantsur @zaneb what are your thoughts?

Reply:
> It seems like we ideally need e.g bmc.AccessDetailsFromDriverInfo

I think it's actually a good idea.

Reply:
@hardys @dtantsur Perhaps we can directly use `accessDetails.BuildBIOSSettings` on the installer side to [process the BIOS](https://github.com/openshift/installer/pull/5196/commits/ed2d50cf68e2fadeab9d547e3aab9052b3b89e07), so there is no need to transfer new fields.

And the installer itself has created the [accessDetails](https://github.com/openshift/installer/blob/master/pkg/tfvars/baremetal/baremetal.go#L78) and called some of its methods (eg. [DriverInfo](https://github.com/openshift/installer/blob/master/pkg/tfvars/baremetal/baremetal.go#L86)), so it is reasonable and more convenient to process BIOS on the installer side.

Reply:
> It seems like we ideally need e.g bmc.AccessDetailsFromDriverInfo

IMHO, the information provided by [DriverInfo](https://github.com/metal3-io/baremetal-operator/blob/main/pkg/hardwareutils/bmc/access.go#L64)(as well as other methods) is not sufficient to instantiate AccessDetails, unless we modify the existing DriverInfo method.

Considering that the installer has [instantiated accessDetails](https://github.com/openshift/installer/blob/master/pkg/tfvars/baremetal/baremetal.go#L78), it might be better to call [BuildBIOSSettings](https://github.com/metal3-io/baremetal-operator/blob/main/pkg/hardwareutils/bmc/access.go#L86) directly on the installer side to process the BIOS.

Reply:
> IMHO, the information provided by [DriverInfo](https://github.com/metal3-io/baremetal-operator/blob/main/pkg/hardwareutils/bmc/access.go#L64)(as well as other methods) is not sufficient to instantiate AccessDetails

DriverInfo + Driver are sufficient to reconstruct AccessDetails, this is pretty much what AccessDetails is about.